test/jdk/java/net/httpclient/PlainConnectionLockTest.java

```diff
@@ -0,0 +1,382 @@
+        // big enough.
+        int backlog = MANY > 50 ? MANY : 50;
+
+        // create a https server for HTTP/1.1
+        var loopback = InetAddress.getLoopbackAddress();
+        var wrappedHttps1Server = HttpsServer.create(new InetSocketAddress(loopback, 0), backlog);
+        wrappedHttps1Server.setHttpsConfigurator(new TestServerConfigurator(loopback, sslContext));
+        https1Server = HttpTestServer.of(wrappedHttps1Server, serverExecutor);
+        https1Server.addHandler((exchange) -> {
+            if (blockResponse(requestSemaphore, responseSemaphore)) {
+                exchange.sendResponseHeaders(200, 0);
+            } else {
+                exchange.sendResponseHeaders(500, 0);
+            }
+        }, "/PlainConnectionLockTest/");
+        https1Server.start();
+        System.out.println("HTTPS Server started at " + https1Server.getAddress());
+        https1URI = "https://" + https1Server.serverAuthority() + "/PlainConnectionLockTest/https1";
+
+        // create a plain http server for HTTP/1.1
+        var wrappedHttp1Server = HttpServer.create(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), backlog);
+        http1Server = HttpTestServer.of(wrappedHttp1Server, serverExecutor);
+        http1Server.addHandler((exchange) -> {
+            if (blockResponse(requestSemaphore, responseSemaphore)) {
+                exchange.sendResponseHeaders(200, 0);
+            } else {
+                exchange.sendResponseHeaders(500, 0);
+            }
+        }, "/PlainConnectionLockTest/");
+        http1Server.start();
+        System.out.println("HTTP Server started at " + http1Server.getAddress());
+        http1URI = "http://" + http1Server.serverAuthority() + "/PlainConnectionLockTest/http1";
+    }
+
+    @AfterEach
+    synchronized void afterTest() throws Exception {
+        if (http1Server != null) {
+            System.out.println("Stopping HTTP server " + http1Server.getAddress());
+            http1Server.stop();
+        }
+        if (https1Server != null) {
+            System.out.println("Stopping HTTPS server " + https1Server.getAddress());
+            https1Server.stop();
+        }
+        if (serverExecutor != null) {
+            System.out.println("Closing server executor");
+            if (successfulCompletion) {
+                serverExecutor.close();
+            } else {
+                // server handlers may be wedged.
+                serverExecutor.shutdownNow();
+            }
+        }
+        requestSemaphore = null;
+        responseSemaphore = null;
+        serverExecutor = null;
+        http1Server = null;
+        https1Server = null;
+        http1URI = null;
+        https1URI = null;
+        System.out.println("done\n");
+    }
+
+    @Test
+    void sendManyHttpRequestsNoShutdown() throws Exception {
+        try {
+            sendManyRequests(http1URI, MANY, false);
+        } catch (Throwable t) {
+            t.printStackTrace(System.out);
+            throw t;
```

Comment:
I see a longing for TestNG, where stack traces were dumped into stdout, and stderr is populated with more verbose logging. :stuck_out_tongue_winking_eye: